Adds RP version storage account

m1kola opened the pull request:
### Which issue this PR addresses:

Fixes [work item #7589988](https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/7589988/)

### What this PR does / why we need it:

It adds a storage account to the global infrastructure where we are going to store currently running RP versions.

Writing data into the account is a separate story.

### Test plan for issue:

I did test deployment of `deploy/rp-production-global.json` template by removing unrelated resources and parameters.

### Is there any documentation that needs to be updated for this PR?

Don't think so: manifest is used in the tooling only.

jim-minter commented on deploy/rp-production-global.json:
```diff
@@ -27,6 +27,16 @@
+                "name": "Standard_LRS"
+            },
+            "location": "[resourceGroup().location]",
+            "name": "arorpversion",
```
storage account names are globally unique.  Need to parameterise this so that it works in INT and PROD - the actual value should come from the RP config repo.

m1kola commented:
Ready for review. There is a separate PR in the RP-Config repo to update configs.

m1kola commented:
Rebased to fix conflicts

mjudeikis commented:
@m1kola this is stale code now.
Apart of this we are missing deployer part code changes, where we define value in `pkg/deploy/config` and pass it in from our coding repository.

m1kola commented:
Rebased on top of the master branch. Added a field into `pkg/deploy/config.go` to support unmarshalling (we do not use it anywhere at the moment).

@mjudeikis please take another look.

m1kola commented:
> pass it in from our coding repository.

Not sure what you mean here. If it is a typo and should be "config repository" - then there is a separate PR in the config.

mjudeikis commented:
> > pass it in from our coding repository.
> 
> Not sure what you mean here. If it is a typo and should be "config repository" - then there is a separate PR in the config.

Check this and how these are wired into deployment 
https://github.com/Azure/ARO-RP/blob/master/pkg/deploy/config.go

asalkeld commented:
This seems like a very strange idea to me. Storing the RP versions in a storage account instead of sending the version as a metric at startup. Would be good to see the rational somewhere - why are we doing it this way and what else was considered.

mjudeikis commented:
> This seems like a very strange idea to me. Storing the RP versions in a storage account instead of sending the version as a metric at startup. Would be good to see the rational somewhere - why are we doing it this way and what else was considered.

Metric is not accessible in the places where we will need it. Most of CI will not be able to access metrics when running tests.
The problem we are solving is that we are running latest tests on any production cluster. We need t be able to match tests with what is in production. Rest API is not possible as RP is secured with mutual TLS and CI will never have access to those certs.
Metrics system is secured too behind SAW. 
There are not many other options here :/
